spec/src/main/asciidoc/architecture.adoc

```diff
@@ -212,17 +212,19 @@ to prevent hard to spot copy & paste errors, where for example all methods of a
 `@Timed(name="myApp", absolute=true)`.
 
 If this behaviour is required, then it is possible to mark all such instances as _reusable_ by passing
-the respective flag in the Annotation. Gauges are not reusable.
+the respective flag in the Annotation. Gauges are never reusable.
+
+NOTE: Reusability concepts are generally meant to apply only to annotated metrics. These restrictions are generally not placed on programmatically registered metrics,
+unless a metric is programmatically registered with the `reusable` flag set to false (which has to be set explicitly).
 
 For metrics created programmatically (by calling methods of the `MetricRegistry`), reusing is allowed by default, so
 multiple calls retrieving an instance of a metric from the registry will return the same metric object so that the object can be reused in
-multiple places in the application.
+multiple places in the application's codebase, and all calls that update the metric's value will update the same metric instance.
 
-The implementation must throw an 'IllegalArgumentException' during a metric registration call when the call would result
-in the reuse of a metric where that metric was either previously declared not reusable or where the registration call itself
-declares the metric to not be reusable.
+The implementation must throw an `IllegalArgumentException` if it detects multiple annotations referring to the same metric without being marked reusable,
+or when there is a call to retrieve a metric which was registered as not reusable.
```

Comment:
"a call to retrieve a metric" is a bit open ended.  For example, an exporter would be expected to retrieve the metrics, but we wouldn't throw an exception when simply getting the list of counters from the MetricsRegistry and iterating through them.

Reply:
Hmm true. And an application calling `registry.getMetrics()` can also retrieve that metric, because it will be in the resulting map. It's a bit tough to describe that this concerns only the get-or-create methods. I'd suggest we don't overcomplicate this and just remove that line (225), would that be ok?

Reply:
"registration call itself declares the metric to not be reusable" -> what is that actually? That could happen if a registration call contains a Metadata with reusable=false, but then if the already existing metric is reusable=true, then it's not valid, because the metadata isn't the same.

Reply:
@donbourne I removed line 225 as we agreed on the call